en/resources/tutorials/codablerouting.md

```diff
@@ -0,0 +1,172 @@
+
+## Building a Create API with a Codable Route for POST requests
+
+Building a Codable Route for a POST API to implement storing a To Do item requires only 4 lines of code. The first step is to register a handler for POST requests on “/todos”:
+
+```swift
+router.post("/todos", handler: storeHandler)
+```
+
+and then provide an implementation of the storeHandler: 
+
+```swift
+func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
+    todoStore.append(todo)
+    completion(todo, nil)
+}
+```
+
+Here the store handler accepts a todo item of type ToDo and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an Error in the the case of failure.
+
+The input parameter and the response parameter in completion handler can be any Swift type you want, as long as it conforms to Codable. For example, we could use the following as our ToDo type:
+
+```swift
+public struct ToDo : Codable {
+    public var title: String        // title of the todo item
+    public var order: Int           // order to display the todo item
+    public var completed: Bool      // has the item been completed
+}
+```
+
+At this point the server has stored the item, but the client has no way of referring to that specific item on the server if it wants to get that item, update it, or delete it. In order to do that, some kind of unique identifier is required.
+
+Returning an identifier to the client for the stored data is typically done in one of two ways:
+
+### 1. Adding an additional Optional field to the type
+Here the ToDo struct would be updated to add an extra field:
+```swift
+public struct ToDo : Codable {
+    public var title: String       
+    public var order: Int      
+    public var completed: Bool 
+    public var id: Int?         // Additional field for identifier
+}
+```
+
+And the storeHandler function would then set it:
+
+```swift
+func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
+    var todo = todo
+    let id = todoStore.count
+    todo.id = id
+    todoStore.append(todo)
+    completion(todo, nil)
+} 
+```
+
+### 2. Responding with an additional Identifier value
+Alternatively you can choose to respond with an additional value in the completion handler that conforms to the Identifier protocol, which String and Int have been extended to conform to already. The additional identifier value is then written into the “Location” header of the response.
+
+Here the storeHandler would become the following:
+```swift
+func storeHandler(todo: ToDo, completion: (Int?, ToDo?, RequestError?) -> Void ) -> Void {
+    var todo = todo
+    let id = todoStore.count
+    todoStore.append(todo)
+    completion(id, todo, nil)
+}
+```
+
+## Building a Read API with a Codable Route for GET requests
```

Comment:
Similar to previous comment about the heading being a bit of a mouthful to say in one go.